File: kafkakewl-common/src/main/scala/com/mwam/kafkakewl/common/persistence/PersistentStore.scala
/*
 * SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.mwam.kafkakewl.common.persistence

import com.mwam.kafkakewl.domain.*
import zio.*
import zio.stream.*

trait PersistentStore {
  def loadLatest(): Task[TopologyDeployments]
  def save(topologyDeployments: TopologyDeployments): Task[Unit]
  def stream(compactHistory: Boolean): Stream[Throwable, TopologyDeployments]
}

File: kafkakewl-common/src/main/kotlin/com/mwam/kafkakewl/common/persistence/PersistentStore.kt
/*
 * SPDX-FileCopyrightText: 2025 Marshall Wace <opensource@mwam.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.mwam.kafkakewl.common.persistence

import com.mwam.kafkakewl.domain.TopologyDeployments
import kotlinx.coroutines.flow.Flow

interface PersistentStore {
    fun loadLatest(): Result<TopologyDeployments>
    fun save(topologyDeployments: TopologyDeployments): Result<Unit>
    fun stream(compactHistory: Boolean): Flow<TopologyDeployments>
}
